Backend/src/routes/api.js

```diff
@@ -47,32 +52,55 @@ client.on("error", (err) => {
  */
 function unpackData(data) {
   let buffOffset = 0; // Byte offset for the buffer array
-  let dataType = ""; // Data type specified in the data format json
+  const xAxisCap = 25; // The max number of data points to have in each array at one time
+  let timestamps = solarCarData["timestamps"]; // The array of timestamps for each set of data added to solarCarData
+
+  // Add the current timestamp to timestamps, limit its length, and update the array in solarCarData
+  timestamps.unshift(timestamp);
+  timestamp ++;
+  if(timestamps.length > xAxisCap) {
+    timestamps.pop();
+  }
+  solarCarData["timestamps"] = timestamps;
 
   for (const property in DATA_FORMAT) {
+    let dataArray = []; // Holds the array of data specified by property that will be put in solarCarData
+    let dataType = ""; // Data type specified in the data format
+
+    if(solarCarData.hasOwnProperty(property)) {
+      dataArray = solarCarData[property];
+    }
     dataType = DATA_FORMAT[property][1];
 
-    // Add the data from the buffer to SOLAR_CAR_DATA
+    // Add the data from the buffer to solarCarData
     switch (dataType) {
       case "uint8":
-        solarCarData[property] = data.readUInt8(buffOffset);
+        // Add the data to the front of dataArray
+        dataArray.unshift(data.readUInt8(buffOffset));
         break;
       case "float":
-        solarCarData[property] = data.readFloatBE(buffOffset);
+        // Add the data to the front of dataArray
+        dataArray.unshift(data.readFloatLE(buffOffset));
         break;
       case "char":
-        solarCarData[property] = String.fromCharCode(
-          data.readUInt8(buffOffset)
-        );
+        // Add the data to the front of dataArray
+        dataArray.unshift(String.fromCharCode(data.readUInt8(buffOffset)));
         break;
       case "bool":
-        solarCarData[property] = Boolean(data.readUInt8(buffOffset));
+        // Add the data to the front of dataArray
+        dataArray.unshift(Boolean(data.readUInt8(buffOffset)));
         break;
       default:
```

Comment:
Should we add something that happens here?

Reply:
Could you please elaborate? Is this in reference to the default case?

Reply:
Yeah right now nothing happens, should we log or warn or something here?

Reply:
Yeah. I added a log for unexpected data types